src/main/kotlin/io/hemin/wien/builder/validating/podcast/ValidatingPodcastGooglePlayBuilder.kt

```diff
@@ -32,14 +32,15 @@ internal class ValidatingPodcastGooglePlayBuilder : PodcastGooglePlayBuilder {
     override fun imageBuilder(imageBuilder: HrefOnlyImageBuilder?): PodcastGooglePlayBuilder = apply { this.imageBuilder = imageBuilder }
 
     override fun build(): Podcast.GooglePlay? {
+        val categories = categoryBuilders.mapNotNull { it.build() }
         if (categories.isEmpty() && allNull(author, owner, description, explicit, block, imageBuilder)) {
             return null
         }
 
         return Podcast.GooglePlay(
             author = author,
             owner = owner,
-            categories = immutableCopyOf(categories),
+            categories = categories,
```

Comment:
This is something that I've missed in the previous PRs. The model classes are immutable by design (at least as immutable as it can get on the JVM). My intention was that the data structures produced by the parser are safe for concurrent access, i.e. they can be used by concurrency models that require immutable messages (e.g. the [Actor model](https://en.wikipedia.org/wiki/Actor_model); [Akka](https://akka.io) being the best example on the JVM). This is also why I chose the builder pattern: The builders are mutable, the resulting models are not.

Hence, when we instantiate model objects from the builders, we should keep producing immutable lists. As we have it right now, it can be considered an indirect break of the API. In one of my personal use cases of the library for example, the compiler would not complain, but weird concurrency issues would arise. I haven't migrated to the current version yet, so it slipped my mind and I haven't noticed.

I did not document this design choice in the README so far, hence you couldn't know this. I'm sorry about this lack of clarity.

Reply:
Yeah I got that was the intention (see the comment in `build.gradle.kts` about eventually replacing Guava with Kotlin's own immutable collections API, which should happen at some point IIRC). The reason why I removed that call to `immutableCopyOf` is that we aren't assigning the `mutableListOf()` to the field anymore, but rather it's the result of `map`ping that mutable list. It feels to me it's not necessary anymore to do a defensive copy, since the `map`ping is essentially already doing the same.

Once `build()` is called, the data models are built with a brand new `List` instance (generally backed by an `ArrayList`, but only the read-only `List` view is available in the models) created by the `mapNotNull()` operation. At this point, the mutability is not an issue anymore: nobody has access to the (potentially, but not necessarily) mutable implementation of that `List`.

Reply:
Ah, you have a good point there. Of course, Kotlin's `List` interface does not support mutating operations. My concerns are then obsolete.

Reply:
Example: https://pl.kotl.in/iZreuZbP9